grakn-graph/src/main/java/ai/grakn/graph/internal/ConceptLog.java

```diff
@@ -216,4 +227,16 @@ boolean isTypeCached(TypeLabel label){
         //noinspection unchecked
         return (X) typeCache.get(label);
     }
+
+    void addedInstance(TypeLabel name){
+        instanceCount.compute(name, (key, value) -> value == null ? 1 : value + 1);
+        cleanupInstanceCount(name);
```

Comment:
Yep. If you do a large deletion then that may mean you will not need to do the super node split. So it's important to track these numbers.